src/Array.mo

```diff
@@ -207,6 +213,10 @@ module {
     sortInPlace(temp, compare);
     freeze(temp)
   };
+  public func sortByLessThanOrEqual<X>(array : [X], isLessThanOrEqual : (X, X) -> Bool) : [X] {
```

Comment:
Is this still an isse with the revised prompt? The bot was having trouble with defining functions returning Order.order because the prompt was actually misleadingly broken.

Maybe we could call `lessThanOrEqual` `totalOrder`?